File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/ui/NonScrollableRecyclerView.java
package com.originb.inkwisenote2.modules.smartnotes.ui;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

public class NonScrollableRecyclerView extends RecyclerView {
    public NonScrollableRecyclerView(@NonNull Context context) {
        super(context);
    }

    public NonScrollableRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public NonScrollableRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent e) {
        // Prevent gesture-based scrolling
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent e) {
        // Prevent gesture-based scrolling
        return false;
    }
}

File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/ui/NonScrollableRecyclerView.kt
package com.originb.inkwisenote2.modules.smartnotes.ui

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.recyclerview.widget.RecyclerView

class NonScrollableRecyclerView : RecyclerView {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onInterceptTouchEvent(e: MotionEvent?): Boolean {
        // Prevent gesture-based scrolling
        return false
    }

    override fun onTouchEvent(e: MotionEvent?): Boolean {
        // Prevent gesture-based scrolling
        return false
    }
}
